resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -407,8 +525,46 @@ private[spark] class Client(
     // TODO be resilient to node failures and try all of them
     val node = kubernetesClient.nodes.list.getItems.asScala.head
     val nodeAddress = node.getStatus.getAddresses.asScala.head.getAddress
-    val url = s"http://$nodeAddress:$servicePort"
-    HttpClientUtil.createClient[KubernetesSparkRestApi](uri = url)
+    val urlScheme = if (driverLaunchSslOptions.enabled) {
+      "https"
+    } else {
+      logWarning("Submitting application details and local jars to the cluster" +
+        " over an insecure connection. Consider configuring SSL to secure" +
```

Comment:
I think we should mention "secret" in the application details, and maybe "Strong consider" turning on SSL